### PR TITLE
Perform basic grid search in experiment.py and standardize logging across involved scripts

### PR DESCRIPTION
**Pull request by gecock**
Aims to satisfy https://github.com/IBM/watson-stt-wer-python/issues/27 by allowing `experiment.py` to loop through STT settings and run transcription experiments for each configuration. 

1. The updates to the README should hopefully explain how `experiments.py` works now. If not, feedback welcome and I'll make changes to README. 
1. Standardized logging for `experiments.py`, `transcribe.py` and `analyze.py`. I took some liberties deciding what log level the existing `print()` lines should be, so again feedback welcome. 
1. Added two command line parameter flags to `experiments.py`, `transcribe.py` and `analyze.py`. I matched the one common flag `-c`/`--config_file` with what exists in `models.py` today. 

Each experiment outputs the following with the default log level of `INFO`:
```
2022-12-21 16:09:50,860 - INFO - Running Experiment -- Character Insertion Bias: 0.0, Customization Weight: 0.0, Speech Detector Sensitivity: 0.5, Background Audio Suppression: 0.0
2022-12-21 16:09:58,324 - INFO - Completed transcribing 4 files out of 4
2022-12-21 16:09:58,326 - INFO - Wrote transcriptions for 4 audio files to ./sample-files/0.0_0.0_0.5_0.0/stt_transcriptions.csv
2022-12-21 16:09:58,336 - INFO - Updated ./sample-files/0.0_0.0_0.5_0.0/stt_transcriptions.csv with reference transcriptions
2022-12-21 16:09:58,338 - INFO - Wrote detailed results to ./sample-files/0.0_0.0_0.5_0.0/sample-details.csv
2022-12-21 16:09:58,339 - INFO - Wrote summary results to ./sample-files/0.0_0.0_0.5_0.0/sample-summary.json
2022-12-21 16:09:58,339 - INFO - Wrote word accuracy results to ./sample-files/0.0_0.0_0.5_0.0/sample-accuracy.csv
2022-12-21 16:09:58,339 - INFO - Experiment Complete 
```

Note that `2022-12-21 16:09:58,324 - INFO - Completed transcribing 4 files out of 4` is a status message dependent on the number of files to transcribe. The script is configured to print out a status message every 100 files and at the final completion of all files. e.g. if there are 410 files there will be 5 messages total (4 in progress messages and the final total)

DCO 1.1 Signed-off-by: Gregory Ecock - [gecock@ibm.com](mailto:gecock@ibm.com)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,6 +113,40 @@ Repo of the Python module JIWER: https://pypi.org/project/jiwer/
...
+
+```
+python analyze.py --config_file config.ini --log_level DEBUG
+```
+
+# Experimenting
+Use the `experiment.py` script to execute a series of Transcription/Analyze experiments to optimize SpeechToText parameters. 
+
+## Setup
+
```
Comment (tnixa) [inline README.md:126]: Sorry if i missed it but do you have the `-c` and `-ll` flag options documented somewhere?

Comment (tnixa) [inline README.md:126]: Sorry if i missed it but do you have the different log level options documented somewhere? Or are we expecting the users to know?

Comment (tnixa) [inline README.md:126]: Would it make any sense to get the log level from the config file instead of a script execution flag?

**Review by tnixa — CHANGES_REQUESTED** [review]
Nice changes! See my comments as I had a few questions...

File: README.md (hunk 1 above)
Comment (gecock) [reply README.md:140]: added a separate section to explain since they are common parameters and then linked to it

Comment (gecock) [reply README.md:140]: added a separate section to explain since they are common parameters and then linked to it

Comment (gecock) [reply README.md:140]: I thought about doing that but landed on the side of cmdline bc changing the log level is something that I'd want to do at runtime without having to go edit any other files. I can toggle it in the terminal which is more convenient if I'm debugging. I can be swayed though, either way would work.

Comment (marconoel) [reply README.md:140]: for debugging, I agree with @gecock : I would rather do it at the cmdline as needed instead of changing the config file.

**Review by marconoel — APPROVED** [review]
I approve the changes

File: README.md (hunk 1 above)
Comment (marconoel) [reply README.md:140]: @tnixa Please confirm this is resolved to your satisfaction and resolve conversation

Comment (marconoel) [reply README.md:140]: @tnixa Please confirm this is resolved to your satisfaction and resolve conversation

Comment (marconoel) [reply README.md:140]: @tnixa Please confirm this is resolved to your satisfaction and resolve conversation

**Review by tnixa — APPROVED** [review]

**Pull request merged**